packages/hoppscotch-backend/src/auth/helper.ts

```diff
@@ -117,11 +118,107 @@ export function authProviderCheck(
 
   const envVariables = VITE_ALLOWED_AUTH_PROVIDERS
     ? VITE_ALLOWED_AUTH_PROVIDERS.split(',').map((provider) =>
-        provider.trim().toUpperCase(),
-      )
+      provider.trim().toUpperCase(),
+    )
     : [];
 
   if (!envVariables.includes(provider.toUpperCase())) return false;
 
   return true;
 }
+
+/**
+ * Extract auth tokens from cookie headers of a request
+ * @param headers HTTP request headers containing auth tokens
+ * @returns AuthTokens for JWT strategy to use
+ */
+export const extractAuthTokensFromCookieHeaders = (headers: IncomingHttpHeaders) => {
+  const cookieHeader = headers['cookie'] || headers['Cookie'] || headers['COOKIE'];
+
+  if (!cookieHeader) {
+    throw new HttpException(COOKIES_NOT_FOUND, 400, {
+      cause: new Error(COOKIES_NOT_FOUND),
+    });
+  }
+
+  const cookieStr = Array.isArray(cookieHeader) ? cookieHeader[0] : cookieHeader;
+
+  const cookies = cookieStr.split(';')
+    .map(cookie => cookie.trim())
+    .reduce((acc, curr) => {
+      const [key, value] = curr.split('=');
+      acc[key] = value;
+      return acc;
+    }, {} as Record<string, string>);
```

Comment:
Can we take out cookies object to map conversion into a separate function? So that the other function `extractAccessTokensFromCookieHeaders` can re-use the logic instead of code duplication.

Reply:
Makes sense, definitely 👍